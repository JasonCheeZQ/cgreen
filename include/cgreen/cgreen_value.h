#ifndef CGREEN_VALUE_HEADER
#define CGREEN_VALUE_HEADER

#include <stdint.h>

typedef enum {INTEGER, STRING, DOUBLE, POINTER} CgreenValueType;

typedef struct {
    CgreenValueType type;
    union {
        intptr_t integer_value;
        double double_value;
        void *pointer_value;
        const char *string_value;
    } value;
} CgreenValue;

#ifdef __cplusplus
namespace cgreen {
#endif

extern CgreenValue make_cgreen_integer_value(intptr_t integer);
extern CgreenValue make_cgreen_string_value(const char *string);
extern CgreenValue make_cgreen_double_value(double value);
extern CgreenValue make_cgreen_pointer_value(void *ptr);

#ifdef __cplusplus
} // namespace cgreen
#endif
 
#endif
