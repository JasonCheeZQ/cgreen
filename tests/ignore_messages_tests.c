#include <cgreen/cgreen.h>

#include <stdio.h>
#include <stdlib.h>
#include <unistd.h>

#ifdef __cplusplus
using namespace cgreen;
#endif

Describe(IgnoreMessage);
BeforeEach(IgnoreMessage) {}
AfterEach(IgnoreMessage) {}

Ensure(IgnoreMessage, should_not_count_empty_tests_as_ignored) {
}

Ensure(IgnoreMessage, should_not_count_passing_tests_as_ignored) {
}

Ensure(IgnoreMessage, should_not_count_failing_tests_as_ignored) {
}

Ensure(IgnoreMessage, should_not_count_exceptions_as_ignored) {
    int segv = *((int*)1);
    (void)segv;                 /* UNUSED */
}

xEnsure(IgnoreMessage, should_present_this_as_ignored) {
    fail_test("Should not be run since it's inside an ignored test");
}


