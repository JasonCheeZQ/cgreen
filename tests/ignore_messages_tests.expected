Running "ignore_messages_tests" (5 tests)...
Completed "IgnoreMessage":  in 0ms.
Completed "ignore_messages_tests":  in 0ms.
